accumulator/pollardutil_test.go

```diff
@@ -0,0 +1,45 @@
+package accumulator
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestPollardSerializeDeserialize(t *testing.T) {
+	var p, q Pollard
+	// generate slice of leaf
+	leaves := make([]Leaf, 10)
+	for i := 0; i < len(leaves); i++ {
+		leaves[i].Hash[0] = uint8(i + 1)
+	}
+	// add leaves to pollard
+	err := p.add(leaves)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// new pollard
+	err = q.add(leaves)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// performing serialization
+	old_byte, err := p.Serialize()
+	if err != nil {
+		t.Fatal(err)
+	}
+	// perform Deserialize
+	err = q.Deserialize(old_byte)
```

Comment:
Actually, the more I think about it, this call to `Deserialize` should throw an error.  The fact that it doesn't right now seems like a bug.

The reason is: this test is supposed to test serialize / deserialize.  But if `Deserialize` did absolutely nothing, this test would still pass.  Because p and q already have the same hashes in them.

I think a check like 
```if numleaves != 0 {
return fmt.Errorf("can't deserialize into populated pollard")
}
```
would fix this and avoid the problem.